### DISCUSSION TITLE
Make the headers collection not throw on non-existent keys

### DISCUSSION THREAD

**Discussion by ZzZombo** [mitmproxy/mitmproxy#6028] [General]
#### Problem Description
I do not understand why is it made so, it's just not convenient to have to wrap every access either in existence checks, error handlers, or use `get_all` and then do the concatenation yourself.

#### Proposal
Return `None` if the key doesn't exist.

#### Alternatives
See above.

#### Additional context
None.

**Comment by mhils** [comment]
As mentioned in the docs, headers mostly behave like a normal dict: You can call `headers.get("key")` or  `headers.get("key", "default")` if you don't want them to raise.
